app/src/app/app.component.ts

```diff
@@ -99,42 +108,63 @@ export class AppComponent implements AfterViewInit {
         });
     }
 
-    public theme = "light";
-    //toggle the current theme
-    public toggleTheme() {
-        this.setTheme(this.theme == "light"? "dark" : "light");
+    // Toggle the current theme
+    public toggleTheme(): void {
+        // tslint:disable-next-line:triple-equals
+        this.setTheme(this.theme == Theme.LightMode ? Theme.DarkMode : Theme.LightMode);
     }
 
-    public setTheme(theme: string) {
-        this.logger.log("setting theme to", theme);
+    /**
+     * Sets the initial/default theme based on the operating system/browser settings. Called by ngOnInit.
+     */
+    public setDefaultTheme(): void {
+        // The theme property will be set to 'dark' if the operating system or browser is in dark mode, and 'light' if
+        // it is in light mode
+        const mediaQueryList = window.matchMedia('(prefers-color-scheme: dark)');
+        this.theme = mediaQueryList.matches ? Theme.DarkMode : Theme.LightMode;
+
+        // use the addEventListener method of the MediaQueryList object to set up a listener that will be called
+        // whenever the light/dark mode setting changes
+        mediaQueryList.addEventListener('change', (event) => {
+          this.theme = event.matches ? Theme.DarkMode : Theme.LightMode;
+        });
+    }
+
+    public setTheme(theme: string): void {
+        this.logger.log(`Setting theme to ${theme}`);
         this.theme = theme;
         const overlayContainerClasses = this.overlayContainer.getContainerElement().classList;
-        overlayContainerClasses.remove("dark", "light");
+        overlayContainerClasses.remove(Theme.DarkMode, Theme.LightMode);
         overlayContainerClasses.add(this.theme);
-        setCookie("theme", theme, 30);
+        setCookie('theme', theme, 30);
+    }
+
+    ngOnInit(): void {
+      this.setDefaultTheme();
     }
```

Comment:
I recommend moving the `setDefaultTheme()` function call from the `ngOnInit` to the else statement in the constructor that sets the default theme to light mode if the `theme` cookie is not present. Currently, when the Workbench loads, the theme is always set twice: the logic in the constructor is executed first to set the theme based on cookies, and then the `ngOnInit` lifecycle hook executes and overrides the theme based on the operating system/browser settings. If a user has manually chosen a dark or light theme in Workbench with the toggle theme button, that should be honored over their default system theme.